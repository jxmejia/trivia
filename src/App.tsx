import { Global } from "@emotion/react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { QuestionsContextProvider } from "./context";
import globalStyles from "./styles/global";
import { Home, Quiz, Results } from "./views";

function App(): JSX.Element {
  // TODO: Add error boundary https://www.npmjs.com/package/react-error-boundary
  return (
    <>
      <Global styles={globalStyles} />
      <QuestionsContextProvider>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/quiz" element={<Quiz />} />
            <Route path="/results" element={<Results />} />
          </Routes>
        </BrowserRouter>
      </QuestionsContextProvider>
    </>
  );
}

export default App;
