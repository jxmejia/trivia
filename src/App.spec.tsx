import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";
import App from "./App";

expect.addSnapshotSerializer(createSerializer());

it("renders component", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
