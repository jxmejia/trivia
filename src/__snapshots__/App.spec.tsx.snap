// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
.emotion-0 {
  text-align: center;
}

.emotion-1 {
  margin: 50px;
}

.emotion-2 {
  font-weight: bold;
  font-size: 18px;
}

.emotion-3 {
  margin: 25px;
}

.emotion-4 {
  font-size: 14px;
}

.emotion-7 {
  font-size: 14px;
  text-transform: uppercase;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        Welcome to the Trivia Challenge!
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div>
        <span
          class="emotion-4"
        >
          You will be presented with 10 True or False questions.
        </span>
      </div>
      <div>
        <span
          class="emotion-4"
        >
          Can you score 100%?
        </span>
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <button
        class="emotion-7"
        type="button"
      >
        Begin
      </button>
    </div>
  </div>
</div>
`;
